crates/nu-protocol/src/ast/call.rs

```diff
@@ -154,30 +154,10 @@ impl Call {
             })
     }
 
-    pub fn positional_iter_mut(&mut self) -> impl Iterator<Item = &mut Expression> {
-        self.arguments
-            .iter_mut()
-            .take_while(|arg| match arg {
-                Argument::Spread(_) => false, // Don't include positional arguments given to rest parameter
-                _ => true,
-            })
-            .filter_map(|arg| match arg {
-                Argument::Named(_) => None,
-                Argument::Positional(positional) => Some(positional),
-                Argument::Unknown(unknown) => Some(unknown),
-                Argument::Spread(_) => None,
-            })
-    }
-
     pub fn positional_nth(&self, i: usize) -> Option<&Expression> {
         self.positional_iter().nth(i)
     }
 
-    // TODO this method is never used. Delete?
-    pub fn positional_nth_mut(&mut self, i: usize) -> Option<&mut Expression> {
```

Comment:
I'm not sure if we should remove it.  It seems like a useful method and can be used inside plugin.

Reply:
Disagree, we should have a minimal stable interface for the plugins now, and can add things back if there is a proven need.

I think the semantic meaning of mutating `Expression`s of a `Call` is weird (bar `mem::take` but only if proven useful). As a plugin you should only concern yourself with the resulting `Value`s and associated `Span`.

Reply:
I don't believe `Call` is exposed to plugins. Rather, plugins use the `EvaluatedCall` type.